tests/Doctrine/Tests/Common/DataFixtures/Executor/PHPCRExecutorTest.php

```diff
@@ -0,0 +1,202 @@
+                return $callback($dm);
+            }));
+        $purger->expects($this->once())->method('purge');
+
+        $executor->execute(array($fixture), false);
+    }
+
+    public function testExecuteFixtureWithoutPurge()
+    {
+        $dm       = $this->getDocumentManager();
+        $purger   = $this->getPurger();
+        $executor = new PHPCRExecutor($dm, $purger);
+        $fixture  = $this->getMockFixture();
+
+        $fixture->expects($this->once())->method('load')->with($dm);
+        $dm
+            ->expects($this->once())
+            ->method('transactional')
+            ->with($this->isType('callable'))
+            ->will($this->returnCallback(function ($callback) use ($dm) {
+                return $callback($dm);
+            }));
+        $purger->expects($this->never())->method('purge');
+
+        $executor->execute(array($fixture), true);
+    }
+
+    public function testFailedTransactionalStopsPurgingAndFixtureLoading()
+    {
+        $dm        = $this->getDocumentManager();
+        $purger    = $this->getPurger();
+        $executor  = new PHPCRExecutor($dm, $purger);
+        $fixture   = $this->getMockFixture();
+        $exception = new Exception();
+
+        $fixture->expects($this->never())->method('load');
+        $dm->expects($this->once())->method('transactional')->will($this->throwException($exception));
+        $purger->expects($this->never())->method('purge');
+
+        try {
+            $executor->execute(array($fixture), true);
+        } catch (\Exception $caughtException) {
+            $this->assertSame($exception, $caughtException);
+        }
+    }
+
+    /**
+     * @return \Doctrine\Common\DataFixtures\Purger\PHPCRPurger|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private function getPurger()
+    {
+        return $this->getMock('Doctrine\Common\DataFixtures\Purger\PHPCRPurger', array(), array(), '', false);
+    }
+
+    /**
+     * @return \Doctrine\ODM\PHPCR\DocumentManager|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private function getDocumentManager()
+    {
+        $this->loadDocumentManagerClass();
+
+        return $this->getMock(
+            'Doctrine\ODM\PHPCR\DocumentManager',
+            array(
+                'transactional',
+                'flush',
+                'clear',
+            ),
+            array(),
+            '',
+            false
+        );
+    }
+
+    /**
+     * @return \Doctrine\Common\DataFixtures\FixtureInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private function getMockFixture()
+    {
+        return $this->getMock('Doctrine\Common\DataFixtures\FixtureInterface', array(), array(), '', false);
+    }
+
+    /**
+     * Ensures that the {@see \Doctrine\ODM\PHPCR\DocumentManager} class exists
+     */
+    private function loadDocumentManagerClass()
+    {
+
+        if (class_exists('Doctrine\ODM\PHPCR\DocumentManager')) {
```

Comment:
:-1: just a dependency for something I'd mock out anyway.

We're acting only on interfaces in any case